Homework11/WebTableTests.cs

```diff
@@ -0,0 +1,226 @@
+            submitButton.Click();
+
+            // Get text of updated entry.
+            ReadOnlyCollection<IWebElement> gridcell = _driver.FindElements(By.XPath("((//div[@role='row'])[2])/div"));
+            Assert.AreEqual(ExpectedConstants.NewEntryValue, GetTextOfCollectionElements(gridcell));
+        }
+
+        // Test5: Delete first entry.
+        [Test]
+        public void DeleteFirstEntry()
+        {
+            // Get the first name from the first row.
+            var firstName1 = _driver.FindElement(By.XPath("((//div[@role='rowgroup'])[1]//descendant::div[@role='gridcell'])[1]")).Text;
+
+            // Remove the first row.
+            var deleteButton = _driver.FindElement(By.XPath("//span[@id='delete-record-1']"));
+            deleteButton.Click();
+
+            // Get the rows left after the delete operation.
+            ReadOnlyCollection<IWebElement> rows = _driver.FindElements(By.XPath("//div[@role='rowgroup']"));
+
+            // Get the rows from the collection above that contains the first name from removed entry.
+            var rowsWithRemovedValue = GetListOfFilteredRows(GetListOfNotEmptyRows(rows), firstName1);
+            Assert.AreEqual(0, rowsWithRemovedValue.Count);
+        }
+
+        [OneTimeTearDown]
+        public void OneTimeTearDown()
+        {
+            _driver.Quit();
+        }
+
+        public void DeleteValue(IWebElement element)
+        {
+            element.SendKeys(Keys.Control + "a" + Keys.Delete);
+        }
+
+        public void FillInInput(IWebElement element, string value)
+        {
+            element.SendKeys(value);
+        }
+
+        private string GetTextOfCollectionElements(ReadOnlyCollection<IWebElement> selectedItems)
+        {
+            var stringBuilder = new StringBuilder();
+            for (int i = 0; i < selectedItems.Count; i++)
+            {
+                IWebElement element = selectedItems[i];
+                string text = element.Text;
+                stringBuilder.Append($"{text} ");
+            }
+            var finalString = stringBuilder.ToString();
+
+            return finalString;
+        }
+
+        private List<string> GetListOfNotEmptyRows(ReadOnlyCollection<IWebElement> selectedRows)
+        {
+            List<string> rowsText = new List<string>();
+            for (int i = 0; i < selectedRows.Count; i++)
+            {
+                var stringBuilder = new StringBuilder();
+                {
+                    IWebElement element = selectedRows[i];
+                    string text = element.Text;
+                    stringBuilder.Append($"{text} ");
+                }
+                var finalString = stringBuilder.ToString();
+                if (finalString != "        ")
+                {
+                    rowsText.Add(finalString);
+                }
+            }
+
+            return rowsText;
+        }
+
+        private List<string> GetListOfFilteredRows(List<string> listOfRows, string filterValue)
```

Comment:
Ok, works. But I think it's to complicated as you need only count of elements. You could do it in test
listOfRows.Count(x => x.Contains(filterValue))

Reply:
yep, it just to show another way)